### ISSUE TITLE
Create Task button not working

### ISSUE THREAD

**Issue by aamirtestable** [Worklenz/worklenz#59]
I have used docker and changed the ports. Frontend gave error of compilation and I added following

```
RUN apk add --no-cache python3 py3-pip make g++
```
in docker file. Right now it is on my local sytem and I have created different members, projects etc. The problem is I m not able to create a task. when I click create task it just start spinning and button is disabled. Nothing happens no request goes to server and I can still navigate to different parts of the app and can make changes. 

edit ---

I tried Create task from board view or Task list view none of them are working. There is only one task which was created from on boarding screens.

**Comment by chamikaJ** [comment]
To create tasks and update various task attributes, such as the end date, status, and priority, we use WebSockets instead of REST API calls. The issue you are encountering might be due to the WebSocket connection not being properly established. 

Could you check your browser console for any errors?

**Comment by aamirtestable** [comment]
Right, I do have lots of dockers and my port 4200 and 3000 are already busy with ember and react. So I have changed the ports to 4300 and 5421. 

I assume ports are working fine as it let me login and also created the first task via on boarding process. 

Now coming to WS. I do have 

```js
const config: SocketIoConfig = {
  url: environment.production ? '' : 'ws://localhost:5421',
  options: {
    transports: ['websocket'],
    path: "/socket",
    upgrade: true
  }
};
```

in my app.module.ts file. I did search all the instances of 4200 port and 3000 ports and changed them. The requests to WS are going to `ws://localhost:5421/socket/?EIO=4&transport=websocket` but they are not returning anything

In docker logs I can confirm when frontend `4300` is hit, the backend server on `5421` is responding

```
2024-12-12 12:29:59 GET /api/v1/settings/notifications 200 204.029 ms - 186
2024-12-12 12:29:59 GET /api/v1/job-titles?index=1&size=5&search= 200 188.424 ms - 76
2024-12-12 12:29:59 GET /api/v1/teams/invites 200 235.784 ms - 57
2024-12-12 12:29:59 GET /secure/verify 200 169.541 ms - 787
2024-12-12 12:29:59 GET /api/v1/home/personal-tasks 200 234.320 ms - 57
2024-12-12 12:29:59 GET /api/v1/teams 200 198.033 ms - 226
2024-12-12 12:29:59 GET /api/v1/home/team-projects 200 48.795 ms - 140
2024-12-12 12:29:59 GET /api/v1/home/tasks?group_by=0&current_tab=All&is_calendar_view=false&selected_date=undefined&time_zone=Asia/Karachi 200 20.691 ms - 950
2024-12-12 12:29:59 GET /api/v1/clients?index=1&size=5&search= 200 12.423 ms - 76
2024-12-12 12:29:59 GET /api/v1/project-categories 200 31.877 ms - 57
2024-12-12 12:29:59 GET /api/v1/home/projects?view=0 200 77.521 ms - 329
2024-12-12 12:30:00 notifications-cron-job: (cron) Notifications job started.
2024-12-12 12:30:00 notifications-cron-job: (cron) Notifications job ended with 0 emails.
2024-12-12 12:30:00 GET /api/v1/job-titles?index=1&size=5&search= 200 19.226 ms - 76
2024-12-12 12:30:00 GET /secure/verify 200 28.311 ms - 787
2024-12-12 12:30:00 GET /api/v1/notifications?filter=Unread 200 39.222 ms - 57
2024-12-12 12:30:00 GET /api/v1/notifications/unread-count 200 6.037 ms - 56
```


So I guess WS are not working

**Comment by aamirtestable** [comment]
sorry to add this in the same ticket, but https://worklenz.com/integrations/ i cant seem to open slack integration documentation. We do use Mattermost and each project have its own separate webhook. This works well with GITLAB as Mattmost webhook are dropin replacement of Slack

**Comment by aamirtestable** [comment]
For sake of experiment, i changed back all the ports to 4200/3000 and it seems working fine. 🤔

**Comment by kalinga777** [comment]
> sorry to add this in the same ticket, but https://worklenz.com/integrations/ i cant seem to open slack integration documentation. We do use Mattermost and each project have its own separate webhook. This works well with GITLAB as Mattmost webhook are dropin replacement of Slack

Hi @aamirtestable , The intergrations will be coming soon in our addon section.

**Comment by aamirtestable** [comment]
I have been trying to make it work. These are the things I have done

I have removed PG container and used a remote DB server, tried from localhost to connect to remote DB (I am not running docker on local and just running both from terminal) I can login, and i can create tickets and websocket is responding well 

Deployed that code to server

1. on server created two domains https://work-bg.xxx.xxx as backend
2. created https://work.xxx.xxx as frontend
3. fixed nginx settings and site is working fine. 
4. Changed the ws url to `wss://work-bg.xxx.xxx` 
5. When socket connection established, i got message `401 Unauthorized` 


So same codebase same database running on two different URLs is giving different results. I think there is something needs to be fixed in `session-middleware.ts` file.

When I login I can see an entry in `pg_sessions` table with correct user ID, and also compared it with an entry made from localhost and both are exactly same. 

@chamikaJ

**Comment by ilovefreesw** [comment]
Maybe you can try changing 127.0.0.1 to your server's IP here 🤔: https://github.com/Worklenz/worklenz/blob/main/worklenz-frontend/proxy.config.json

**Comment by aamirtestable** [comment]
when it is run in docker it used proxy-docker.config.json fie

**Comment by aamirtestable** [comment]
Okay here are things which i did and it seems working now

1. removed DB as I have my own self hosted PG server
2. in docker-compose passed all the environemnt varialbes like DATABASE=${DATABASE}
3. docker compose -> HOSTNAME value was not working from .env file so hardcoded it to my `worklenz.xxxx.yyy`
4. copied the env from backend to root near docker-compose.yml
5. created a subdomain in my hosting panel `https://work.xxxx.yyy` and edited the file where added a `location /socket` section. also changed the internal proxy_pass to go to http://127.0.0.1:4200
6. created a new file to open `https://work.xxxx.yyy:8111` and in the nginx conf file added `location /socket` section and also changed the internal port to go to http:127.0.0.1:3000
8. Add `"https://*.xxxx.yyy",` in ` "connect-src":` json in file `worklenz-backend/src/shared/csp.ts`
9. changed ` url: 'wss://work.xxxx.yyy:8111',` in worklenz-frontend/src/app/app.module.ts

and some other changes like this. in session-middleware and src/bin/www.ts file

I hope this might help someone
